src/main/java/com/sublinks/sublinksapi/api/lemmy/v3/user/controllers/UserController.java

```diff
@@ -282,40 +277,90 @@ public LoginResponse saveUserSettings(@Valid @RequestBody SaveUserSettings saveU
+    }
+    if (saveUserSettingsForm.blur_nsfw() != null) {
+      person.setBlurNsfw(saveUserSettingsForm.blur_nsfw());
+    }
+    if (saveUserSettingsForm.auto_expand() != null) {
+      person.setAutoExpanding(saveUserSettingsForm.auto_expand());
+    }
+    if (saveUserSettingsForm.collapse_bot_comments() != null) {
+      person.setCollapseBotComments(saveUserSettingsForm.collapse_bot_comments());
+    }
+    if (saveUserSettingsForm.enable_keyboard_navigation() != null) {
+      person.setKeyboardNavigation(saveUserSettingsForm.enable_keyboard_navigation());
+    }
+    if (saveUserSettingsForm.enable_animated_images() != null) {
+      person.setAnimatedImages(saveUserSettingsForm.enable_animated_images());
+    }
+    if (saveUserSettingsForm.show_bot_accounts() != null) {
+      person.setShowBotAccounts(saveUserSettingsForm.show_bot_accounts());
+    }
+    if (saveUserSettingsForm.show_scores() != null) {
+      person.setShowScores(saveUserSettingsForm.show_scores());
+    }
+    if (saveUserSettingsForm.theme() != null) {
+      person.setDefaultTheme(saveUserSettingsForm.theme());
+    }
+    if (saveUserSettingsForm.default_sort_type() != null) {
+      person.setDefaultSortType(
+          conversionService.convert(saveUserSettingsForm.default_sort_type(), SortType.class));
+    }
+    if (saveUserSettingsForm.default_listing_type() != null) {
+      person.setDefaultListingType(
+          conversionService.convert(saveUserSettingsForm.default_listing_type(),
+              ListingType.class));
+    }
+    if (saveUserSettingsForm.interface_language() != null) {
+      person.setInterfaceLanguage(saveUserSettingsForm.interface_language());
+    }
+    if (saveUserSettingsForm.show_read_posts() != null) {
+      person.setShowReadPosts(saveUserSettingsForm.show_read_posts());
+    }
+    if (saveUserSettingsForm.email() != null) {
+      person.setEmail(saveUserSettingsForm.email());
+    }
+    if (saveUserSettingsForm.open_links_in_new_tab() != null) {
+      person.setOpenLinksInNewTab(saveUserSettingsForm.open_links_in_new_tab());
+    }
+    if (saveUserSettingsForm.show_avatars() != null) {
+      person.setShowAvatars(saveUserSettingsForm.show_avatars());
+    }
+    if (saveUserSettingsForm.send_notifications_to_email() != null) {
+      person.setSendNotificationsToEmail(saveUserSettingsForm.send_notifications_to_email());
+    }
+    if (saveUserSettingsForm.bot_account() != null) {
+      person.setBotAccount(saveUserSettingsForm.bot_account());
+    }
+    if (saveUserSettingsForm.matrix_user_id() != null) {
+      person.setMatrixUserId(saveUserSettingsForm.matrix_user_id());
+    }
+    if (saveUserSettingsForm.discussion_languages() != null) {
+      final List<Language> languages = new ArrayList<>();
+      for (String languageCode : saveUserSettingsForm.discussion_languages()) {
+        final Optional<Language> language = localInstanceContext.languageRepository().findById(
```

Comment:
This seems like unnecessary DB calls. Why not grab all where "id in ()" instead and then loop through the results. This would mean only a single DB call vs potentially hundreds.